permission_handler_platform_interface/lib/src/permissions.dart

```diff
@@ -31,23 +31,30 @@ class Permission {
   /// iOS: Calendar (Events)
   static const calendar = Permission._(0);
 
+  /// Permission to add events to device's calendar (iOS only).
+  ///
+  /// Events can only be added. If you want to read them as well, use the
+  /// [calendar] permission instead.
+  /// iOS: Calendar (17+ read & write access level)
+  static const calendarAddOnly = Permission._(1);
```

Comment:
We prefer adding new permissions at the end of the list and giving them a higher id, instead of shuffling ids around. While your approach is nicer to the maintainers eye, it is usually a source of bugs because we forget to sync the ids between dart and the platform code. Also, we'd need to update all the platform packages too.